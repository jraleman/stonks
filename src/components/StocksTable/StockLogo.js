import React from 'react';
import styled from 'styled-components';

const StyledImage = styled.img`
    height: 50px;
    width: auto;
`;

const StockLogo = ({ src }) => (
    <StyledImage src={src} />
);

export default StockLogo;
