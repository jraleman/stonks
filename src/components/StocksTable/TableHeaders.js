import React from 'react';

const TableHeaders = ({ headers }) => headers.map(column => (
    <th {...column.getHeaderProps()}>
        {column.render('Header')}
    </th>
));

export default TableHeaders;
