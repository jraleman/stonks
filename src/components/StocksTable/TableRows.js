import React from 'react';
import styled from 'styled-components';

const StyledRow = styled.tr`
    cursor: pointer;
`;

const TableRows = ({ rows, prepareRow, onClick }) => rows.map((row) => {
    const handleOnClick = () => onClick(row.original);
    
    prepareRow(row);
    return (
        <StyledRow {...row.getRowProps()} onClick={handleOnClick}>
            {row.cells.map(cell => (
                <td {...cell.getCellProps()}>
                    {cell.render("Cell")}
                </td>
            ))}
        </StyledRow>
    );
});

export default TableRows;
