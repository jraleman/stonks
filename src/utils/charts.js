import * as d3 from 'd3';
import isEmpty from 'lodash.isempty';

// source: https://brendansudol.com/writing/responsive-d3
const responsivefy = (svg) => {
    if (!svg.node()) {
        return;
    }
    // get container + svg aspect ratio
    const container = d3.select(svg.node().parentNode);
    const width = parseInt(svg.style('width'));
    const height = parseInt(svg.style('height'));
    const aspect = width / height;

    // get width of container and resize svg to fit it
    const resize = () => {
        const targetWidth = parseInt(container.style('width'));
        svg.attr('width', targetWidth);
        svg.attr('height', Math.round(targetWidth / aspect));
    }

    // add viewBox and preserveAspectRatio properties,
    // and call resize so that svg resizes on inital page load
    svg.attr('viewBox', `0 0 ${width} ${height}`)
        .attr('perserveAspectRatio', 'xMinYMid')
        .call(resize);

    // to register multiple listeners for same event type, 
    // you need to add namespace, i.e., 'click.foo'
    // necessary if you call invoke this function for multiple svgs
    // api docs: https://github.com/mbostock/d3/wiki/Selections#on
    d3.select(window).on(`resize.${container.attr('id')}`, resize);
};

const movingAverage = (data, numberOfPricePoints) =>
    data.map((row, index, total) => {
        const start = Math.max(0, index - numberOfPricePoints);
        const subset = total.slice(start, index + 1);
        const sum = subset.reduce((a, b) => a + b.close, 0);
        return ({
            date: new Date(row.date),
            average: sum / subset.length
        });
    });

const clearChart = (id) => {
    d3.select(id).select('svg').remove();
};

export const drawLinearChart = ({
    id = '',
    data = [{}],
    xAxis = '',
    yAxis = '',
    xConvert = (x) => x,
    yConvert = (y) => y,
    colors = {
        price: '#4682b4',
        average: '#ffa500',
        gain: '#03a678',
        lose: '#c0392b',
    }
}) => {
    const margin = { top: 50, right: 50, bottom: 50, left: 50 };
    const width = window.innerWidth - margin.left - margin.right;
    const height = window.innerHeight - margin.top - margin.bottom;

    const xMin = d3.min(data, d => xConvert(d[xAxis]));
    const xMax = d3.max(data, d => xConvert(d[xAxis]));
    const yMin = d3.min(data, d => yConvert(d[yAxis]));
    const yMax = d3.max(data, d => yConvert(d[yAxis]));

    const xScale = d3
        .scaleTime()
        .domain([xMin, xMax])
        .range([0, width]);
    
    const yScale = d3
        .scaleLinear()
        .domain([yMin - 5, yMax])
        .range([height, 0]);
    
    const line = d3
        .line()
        .x(d => xScale(xConvert(d[xAxis])))
        .y(d => yScale(yConvert(d[yAxis])));
    
    const movingAverageData = movingAverage(data, 49);

    const movingAverageLine = d3
        .line()
        .x(d => xScale(xConvert(d[xAxis])))
        .y(d => yScale(yConvert(d.average)))
        .curve(d3.curveBasis);
    
    const volData = data.filter(d => d.volume);
    const yMinVolume = d3.min(volData, d => Math.min(d.volume));
    const yMaxVolume = d3.max(volData, d => Math.max(d.volume));
    const yVolumeScale = d3
        .scaleLinear()
        .domain([yMinVolume, yMaxVolume])
        .range([height, 0]);

    clearChart(id);

    // setup
    const svg = d3
        .select(id)
        .append('svg')
        .attr('width', width + margin.left + margin.right)
        .attr('height', height + margin.top + margin.bottom)
        .call(responsivefy)
        .append('g')
        .attr('transform', `translate(${margin.left},  ${margin.top})`);
    // x axis
    svg
        .append('g')
        .attr('id', 'xAxis')
        .attr('transform', `translate(0, ${height})`)
        .call(d3.axisBottom(xScale));
    // y axis
    svg
        .append('g')
        .attr('id', 'yAxis')
        .attr('transform', `translate(${width}, 0)`)
        .call(d3.axisRight(yScale));
    // draw price line
    svg
        .append('path')
        .data([data])
        .style('fill', 'none')
        .attr('id', 'priceChart')
        .attr('stroke', colors.price)
        .attr('stroke-width', '1.5')
        .attr('d', line);
    // draw avg curve
    svg
        .append('path')
        .data([movingAverageData])
        .style('fill', 'none')
        .attr('id', 'movingAverageLine')
        .attr('stroke', colors.average)
        .attr('d', movingAverageLine);
    // draw volume bar
    svg
        .selectAll()
        .data(volData)
        .enter()
        .append('rect')
        .attr('x', d => xScale(xConvert(d[xAxis])))
        .attr('y', d => yVolumeScale(d.volume))
        .attr('fill', (d, i) =>
            (i && volData[i - 1].close > d.close ? colors.lose : colors.gain))
        .attr('width', 1)
        .attr('height', d => (height - yVolumeScale(d.volume)));
    
    return svg;
};

export const getAverageInfo = (stockData) => {
    if (isEmpty(stockData)) {
        return [{}];
    }
    return (
        Array.from(stockData.reduce(
            (acc, obj) => Object.keys(obj).reduce( 
                (acc, key) => typeof obj[key] == "number"
                    ? acc.set(key, (acc.get(key) || []).concat(obj[key]))
                    : acc,
            acc),
            new Map()
        )).reduce(
            (acc, [name, values]) =>
                Object.assign(acc, { [name]: values.reduce((a,b) => a + b) / values.length }),
            {}
    ));
};
