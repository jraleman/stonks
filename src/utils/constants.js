export const navTabs = [
    { title: 'Stocks List', href: '/stocks' },
];
export const navDropdownItems = [
    { label: 'About', onClick: () => window.alert('about') },
    { label: 'Language', onClick: () => window.alert('language') },
];
export const baseURL = 'https://cloud.iexapis.com/stable';
