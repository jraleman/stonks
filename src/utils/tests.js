import { render } from '@testing-library/react';
import StocksProvider from '../context/stocks-context';

export const stocksCtxRender = (ui, { initialState = {}, ...renderOptions }) =>
    render(
        <StocksProvider {...initialState}>
            {ui}
        </StocksProvider>,
        renderOptions
    );
