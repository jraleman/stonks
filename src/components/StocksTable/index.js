import React, { useMemo, useContext } from 'react';
import { useTable } from 'react-table';
import { Table } from 'reactstrap';
import pick from 'lodash.pick';
import { tableColumns, basicInfo } from '../../utils/constants';
import { StocksDispatchContext, StocksStateContext } from '../../context/stocks-context';
import { getChartForStock } from '../../utils/requests';
import TableHeaders from './TableHeaders';
import TableRows from './TableRows';

const StocksTable = ({ stocks }) => {
    const data = useMemo(() => stocks.map((d) => pick(d, basicInfo)));
    const columns = useMemo(() => tableColumns);
    const {
        getTableProps,
        getTableBodyProps,
        rows,
        headers,
        prepareRow,
    } = useTable({
        columns,
        data
    });
    const dispatch = useContext(StocksDispatchContext);
    const { symbol: contextSymbol } = useContext(StocksStateContext);

    const loadChart = async (symbol) => {
        // TODO: user can change range
        const range = '';
        const res = await getChartForStock({ symbol, range });
        const { data: charts } = res || {};
        dispatch({ type: 'LOAD_STOCK', payload: { symbol, charts } });
    }

    const handleOnClick = (row) => {
        const { symbol } = row || {};
        if (contextSymbol === symbol) {
            dispatch({ type: 'RESET_CONTEXT' });
        } else {
            loadChart(symbol);
        }
    };

    return (
        <Table dark {...getTableProps()}>
            <thead>
                <tr>
                    <TableHeaders headers={headers} />
                </tr>
            </thead>
            <tbody {...getTableBodyProps()}>
                <TableRows
                    rows={rows}
                    prepareRow={prepareRow}
                    onClick={handleOnClick}
                />
            </tbody>
        </Table>
    );
};

export default StocksTable;
