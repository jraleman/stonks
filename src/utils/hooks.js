import { useState, useEffect } from 'react';
import { getQuotesForStock, getLogoForStock, getStatus } from './requests';
import { stocksList } from './constants';

// Returns stocks data, and status of server
export const useStocks = () => {
    const [status, setStatus] = useState(false);
    const [stocks, setStocks] = useState([]);

    useEffect(() => {
        const s = getStatus();
        setStatus(s);
    }, [setStatus]);

    useEffect(() => {
        const fetchQuote = async ({ symbol }) => {
            const { data: stock } = await getQuotesForStock({ symbol });
            const { data: { url: logo }} = await getLogoForStock({ symbol });
            const item = { ...stock, logo };
            setStocks((s) => [...s, item])
        };
        const populateStocks = async () => {
            for (let i = 0; i < stocksList.length; i += 1) {
                fetchQuote({ symbol: stocksList[i] });
            }
        }
        if (status) {
            populateStocks();
        }
    }, [status]);

    return [stocks, status];
};
