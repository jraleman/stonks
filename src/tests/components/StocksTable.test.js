import { render } from '@testing-library/react';
import StocksTable from '../../components/StocksTable';
import TableHeaders from '../../components/StocksTable/TableHeaders';
import TableRows from '../../components/StocksTable/TableRows';
import StockLogo from '../../components/StocksTable/StockLogo';
import { stocksCtxRender } from '../../utils/tests';

describe('StocksTable', () => {
    it('should render', () => {
        const stocks = [];
        const initialState = {
            symbol: '',
            stockData: [{}]
        };
        const component = stocksCtxRender(<StocksTable stocks={stocks} />, { initialState });
        expect(component).toMatchSnapshot();
    });
});

describe('StockLogo', () => {
    it('should render', () => {
        const src = '';
        const component = render(<StockLogo src={src} />);
        expect(component).toMatchSnapshot();
    });

    it('uses a source uri', () => {
        const src = 'https://placeholder.com/wp-content/uploads/2018/10/placeholder.com-logo1.jpg';
        const component = render(<StockLogo src={src} />);
        expect(component).toMatchSnapshot();
    });
});

describe('TableHeaders', () => {
    it('should render', () => {
        const headers = [];
        const component = render(<TableHeaders headers={headers} />);
        expect(component).toMatchSnapshot();
    });
});

describe('TableRows', () => {
    it('should render', () => {
        const rows = [];
        const prepareRow = () => { };
        const onClick = () => { };
        const component = render(
            <TableRows
                rows={rows}
                prepareRow={prepareRow}
                onClick={onClick}
            />
        );
        expect(component).toMatchSnapshot();
    });
});

