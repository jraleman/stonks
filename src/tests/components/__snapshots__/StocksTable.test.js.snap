// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockLogo should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class="sc-bdvvaa hpwfml"
        src=""
      />
    </div>
  </body>,
  "container": <div>
    <img
      class="sc-bdvvaa hpwfml"
      src=""
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StockLogo uses a source uri 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        class="sc-bdvvaa hpwfml"
        src="https://placeholder.com/wp-content/uploads/2018/10/placeholder.com-logo1.jpg"
      />
    </div>
  </body>,
  "container": <div>
    <img
      class="sc-bdvvaa hpwfml"
      src="https://placeholder.com/wp-content/uploads/2018/10/placeholder.com-logo1.jpg"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StocksTable should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="table table-dark"
        role="table"
      >
        <thead>
          <tr>
            <th
              colspan="1"
              role="columnheader"
            />
            <th
              colspan="1"
              role="columnheader"
            >
              Symbol
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Open
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Close
            </th>
            <th
              colspan="1"
              role="columnheader"
            >
              Price
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="table table-dark"
      role="table"
    >
      <thead>
        <tr>
          <th
            colspan="1"
            role="columnheader"
          />
          <th
            colspan="1"
            role="columnheader"
          >
            Symbol
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Open
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Close
          </th>
          <th
            colspan="1"
            role="columnheader"
          >
            Price
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableHeaders should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableRows should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
